core/partitioning/partitioning.h

```diff
@@ -18,15 +18,24 @@ typedef std::unordered_map<const torch::jit::Value*, torch::jit::IValue> Example
 typedef std::pair<std::shared_ptr<torch::jit::Graph>, std::unordered_map<torch::jit::Value*, torch::jit::Value*>>
     GraphAndMapping;
 
-ExampleIValues generateRandomInputs(ir::CollectionInputSpecMap& input_ranges, ir::CollectionTypeMap& input_types);
+ExampleIValues generateRandomInputs(
+    ir::CollectionInputSpecMap& input_ranges,
+    ir::CollectionTypeMap& input_types,
+    const ir::ShapeMode& shape_mode = ir::ShapeMode::kOPT);
 
-void runShapeAnalysis(PartitioningCtx* ctx, torch::jit::Block* block, ExampleIValues& ivalues_maps);
+void populateInputIValues(PartitioningCtx* ctx);
```

Comment:
Looks like generateRandomInputs(), populateInputIvalues() are self-contained for PartitioningCtx, what if we make them member functions of PartitioningCtx?

Reply:
I agree with this. However, when I moved generateRandomInputs() as a member function for PartitioningCtx, I faced errors for some testcases "Unable to find input value torch::jitValue for the block....". I believe the input_values map gets recursively populated with individual seg_block outputs and it wasn't getting updated properly. So I had to keep this separate for now.  
Added a TODO: for now https://github.com/pytorch/TensorRT/blob/dyn_shapes/core/compiler.cpp#L143